UI/ChatUI/ChatUI/ChatUI/MainPage.xaml

```diff
@@ -0,0 +1,446 @@
+							</Grid>
+						</ControlTemplate>
+					</Setter.Value>
+				</Setter>
+			</Style>
+
+			<presentation:MessageTemplateSelector x:Key="MessageTemplateSelector">
+				<presentation:MessageTemplateSelector.MyMessageTemplate>
+					<DataTemplate>
+						<StackPanel Margin="20,15,0,0">
+							<Border
+								MinWidth="300"
+								MinHeight="100"
+								Padding="10"
+								Background="DeepPink"
+								CornerRadius="10">
+								<Grid>
+									<Grid.RowDefinitions>
+										<RowDefinition Height="*" />
+										<RowDefinition Height="*" />
+										<RowDefinition Height="Auto" />
+									</Grid.RowDefinitions>
+									<TextBlock FontWeight="Bold" Text="{Binding ContactName}" />
+									<TextBlock
+										Grid.Row="1"
+										Text="{Binding Text}"
+										TextWrapping="WrapWholeWords" />
+									<TextBlock
+										Grid.Row="2"
+										HorizontalAlignment="Right"
+										Text="{Binding UserFriendlyTimestamp}" />
+								</Grid>
+							</Border>
+							<Polygon
+								Margin="0,0,10,0"
+								HorizontalAlignment="Right"
+								Fill="DeepPink"
+								Points="0,0 10,0 10,10"
+								Visibility="{Binding IsLastInSequence, Converter={StaticResource BooleanToVisibility}}" />
+						</StackPanel>
+					</DataTemplate>
+				</presentation:MessageTemplateSelector.MyMessageTemplate>
+				<presentation:MessageTemplateSelector.OtherMessageTemplate>
+					<DataTemplate>
+						<StackPanel Margin="0,15,20,0">
+							<Border
+								MinWidth="300"
+								MinHeight="100"
+								Padding="10"
+								Background="DeepSkyBlue"
+								CornerRadius="10">
+								<Grid>
+									<Grid.RowDefinitions>
+										<RowDefinition Height="*" />
+										<RowDefinition Height="*" />
+										<RowDefinition Height="Auto" />
+									</Grid.RowDefinitions>
+									<TextBlock FontWeight="Bold" Text="{Binding ContactName}" />
+									<TextBlock
+										Grid.Row="1"
+										Text="{Binding Text}"
+										TextWrapping="WrapWholeWords" />
+									<TextBlock
+										Grid.Row="2"
+										HorizontalAlignment="Right"
+										Text="{Binding UserFriendlyTimestamp}" />
+								</Grid>
+							</Border>
+							<Polygon
+								Margin="10,0,0,0"
+								HorizontalAlignment="Left"
+								Fill="DeepSkyBlue"
+								Points="0,0 10,0 0,10"
+								Visibility="{Binding IsLastInSequence, Converter={StaticResource BooleanToVisibility}}" />
+						</StackPanel>
+					</DataTemplate>
+				</presentation:MessageTemplateSelector.OtherMessageTemplate>
+			</presentation:MessageTemplateSelector>
+		</ResourceDictionary>
+	</Page.Resources>
+
+	<Grid>
+		<Grid.RowDefinitions>
+			<RowDefinition Height="*" />
+			<RowDefinition Height="Auto" />
+		</Grid.RowDefinitions>
+		<mvux:FeedView Source="{Binding Messages}"
+					   Grid.Row="0">
+			<mvux:FeedView.ValueTemplate>
+				<DataTemplate>
+					<ListView behaviors:ReversedPointerWheelBehavior.IsReversed="True"
```

Comment:
We should put this directly in the style, so when we will extract the style we won't forget it.

Reply:
Done